apps/api/src/app/events/usecases/process-subscriber/process-subscriber.usecase.ts

```diff
@@ -0,0 +1,128 @@
+    try {
+      subscriber = await this.getSubscriber(command, template._id);
+      notification = await this.createNotification(command, template._id, subscriber.subscriberId);
+    } catch (e) {
+      return {
+        status: 'subscriber_not_found',
+      };
+    }
+
+    const steps = matchMessageWithFilters(template.steps, command.payload);
+    for (const step of steps) {
+      await this.sendMessage.execute(
+        SendMessageCommand.create({
+          identifier: command.identifier,
+          payload: command.payload,
+          step,
+          transactionId: command.identifier,
+          notificationID: notification._id,
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          userId: command.userId,
+          subscriberId: subscriber._id,
+        })
+      );
+    }
+
+    await this.createLogUsecase.execute(
+      CreateLogCommand.create({
+        transactionId: command.transactionId,
+        status: LogStatusEnum.INFO,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        notificationId: notification._id,
+        text: 'Request processed',
+        userId: command.userId,
+        subscriberId: subscriber._id,
+        code: LogCodeEnum.TRIGGER_PROCESSED,
+        templateId: notification._templateId,
+      })
+    );
+
+    return {
+      status: 'success',
+    };
+  }
+
+  private async getSubscriber(command: ProcessSubscriberCommand, templateId: string): Promise<SubscriberEntity> {
+    const subscriberPayload = command.to;
+    const subscriber = await this.subscriberRepository.findOne({
+      _environmentId: command.environmentId,
+      subscriberId: subscriberPayload.subscriberId,
+    });
+
+    if (subscriber) {
+      return subscriber;
+    }
+    if (subscriberPayload.email || subscriberPayload.phone) {
+      return await this.createSubscriberUsecase.execute(
+        CreateSubscriberCommand.create({
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          subscriberId: subscriberPayload.subscriberId,
+          email: subscriberPayload.email,
+          firstName: subscriberPayload.firstName,
+          lastName: subscriberPayload.lastName,
+          phone: subscriberPayload.phone,
+        })
+      );
+    }
+    await this.createLogUsecase.execute(
+      CreateLogCommand.create({
+        transactionId: command.transactionId,
+        status: LogStatusEnum.ERROR,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        text: 'Subscriber not found',
+        userId: command.userId,
+        code: LogCodeEnum.SUBSCRIBER_NOT_FOUND,
+        templateId: templateId,
+        raw: {
+          payload: command.payload,
+          subscriber: subscriberPayload,
+          triggerIdentifier: command.identifier,
+        },
+      })
+    );
+
+    throw new Error('subscriber_not_found');
```

Comment:
I think in case we know a case of exception we should not throw it, it will add a non-essential CPU processing. 
Maybe it will be a good idea to return a boolean or null and validate the extraction of the subscriber on ourside. 
What do you think about it?